cognite/client/_api/datapoints.py

```diff
@@ -646,23 +647,56 @@ def insert_dataframe(self, dataframe, external_id_headers: bool = False):
                 >>> c.datapoints.insert_dataframe(df)
         """
         np = utils._auxiliary.local_import("numpy")
-        assert not dataframe.isnull().values.any(), "Dataframe contains NaNs. Remove them in order to insert the data."
-
-        assert np.isfinite(dataframe.select_dtypes(include=[np.number])).values.all(
-            axis=None
-        ), "Dataframe contains Infinity. Remove them in order to insert the data."
-        dps = []
-        for col in dataframe.columns:
-            dps_object = {
-                "datapoints": list(
-                    zip(dataframe.index.values.astype("datetime64[ms]").astype("int64").tolist(), dataframe[col])
-                )
-            }
-            if external_id_headers:
-                dps_object["externalId"] = col
-            else:
-                dps_object["id"] = int(col)
-            dps.append(dps_object)
+        if dropna:
+            assert not np.isinf(dataframe.select_dtypes(include=[np.number])).values.any(
+                axis=None
+            ), "Dataframe contains Infinity. Remove them in order to insert the data."
+            dps = []
+            for col in dataframe.columns:
+                if np.all(np.isnan(dataframe[col])):
+                    continue
+                dps_object = {
+                    "datapoints": list(
+                        zip(
+                            dataframe[np.isnan(dataframe[col]) == 0]
+                                .index.values.astype("datetime64[ms]")
+                                .astype("int64")
+                                .tolist(),
+                            dataframe[np.isnan(dataframe[col]) == 0][col],
+                        )
+                    )
+                }
+                if external_id_headers:
+                    dps_object["externalId"] = col
+                else:
+                    dps_object["id"] = int(col)
+                dps.append(dps_object)
+            assert dps, "No valid in DataFrame"
```

Comment:
Why do you want to (potentially) raise `AssertionError` here? 🤔

Reply:
Wasn't sure if trying to post an empty dps would throw an error but probably better to let the API throw that?

Reply:
The final call to:
`self.insert_multiple(dps)`
just ignores it, it seems